### PR TITLE
Updating apiary and adding anonymous user chapter

### PR DESCRIPTION
**Pull request by VanDavv**
Documentation updates + anonymous user requests description

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -41,14 +41,12 @@ Study Box REST API backend documentation and mock.
 
 Response with HTTP code different then **2xx** family should be treated as a failure response. 
 By default API return response with HTTP code **4xx** family.
+Error response message is empty, with just http code included.
 
-### Example HTTP 400 Bad request response:
```
Comment (mlotysz) [inline apiary.apib]: I'd like to fix implementation instead.
BTW try to send some invalid POST request and check and document response here (validation).

Comment (VanDavv) [inline apiary.apib]: It's just status code returned for most cases. Only when there is DataAccessException thrown, there is JSON body.

File: apiary.apib [hunk 2]
```diff
@@ -41,15 +43,23 @@ Study Box REST API backend documentation and mock.
 
 Response with HTTP code different then **2xx** family should be treated as a failure response. 
 By default API return response with HTTP code **4xx** family.
+Any response with HTTP code **500** should be reported ASAP to BE team
```
Comment (mlotysz) [inline apiary.apib]: add http 500 json format description

**Comment by mlotysz** [comment]
Thanks!

**Pull request merged**